### PR TITLE
Fwupdtool reinstall

### PR DESCRIPTION
**Pull request by FFY00**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/hughsie/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by superm1** [comment]
Need to add bash completion and fish completion for it

File: src/fu-engine.c [hunk 1]
```diff
@@ -3764,7 +3764,7 @@ fu_engine_add_releases_for_device_component (FuEngine *self,
 	return TRUE;
 }
 
-static GPtrArray *
+GPtrArray *
```
Comment (superm1) [inline src/fu-engine.c]: This should have some gtk-doc if it's being exported now

**Comment by FFY00** [comment]
> Need to add bash completion and fish completion for it

But this is for `fwupdtool`, not `fwupdmgr`.

File: src/fu-tool.c [hunk 2]
```diff
@@ -1133,6 +1133,74 @@ fu_util_update (FuUtilPrivate *priv, gchar **values, GError **error)
...
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INVALID_ARGS,
+				     "Invalid arguments");
+		return FALSE;
+	}
+
+	if (!fu_util_start_engine (priv, FU_ENGINE_LOAD_FLAG_NONE, error))
+		return FALSE;
+
```
Comment (superm1) [inline src/fu-tool.c]: It would be nice if we can support the device prompt `fwupdmgr` has here too, but not sure how hard that is to do.

**Comment by superm1** [comment]
> > Need to add bash completion and fish completion for it
> 
> But this is for `fwupdtool`, not `fwupdmgr`.

We do have bash-completion for fwupdtool at least.  If there isn't fish completion already no need to add it, but should add it for bash for sure.

File: src/fu-tool.c (hunk 2 above)
Comment (FFY00) [reply src/fu-tool.c]: I can copy the code from `fwupdmgr`, I just though the maintenance burden would probably not be worth.

File: data/bash-completion/fwupdtool.in [hunk 3]
```diff
@@ -100,7 +100,7 @@ _fwupdtool()
 			_show_modifiers
 		fi
 		;;
-	attach|detach|activate|verify-update)
+	attach|detach|activate|verify-update|reinstall)
```
Comment (superm1) [inline data/bash-completion/fwupdtool.in]: I think you need to modify `_fwupdtool_cmd_list` too

File: src/fu-tool.c (hunk 2 above)
Comment (superm1) [reply src/fu-tool.c]: Yeah I guess we don't have it in use for the other commands either right now, so big deal.

File: src/fu-engine.c (hunk 1 above)
Comment (FFY00) [reply src/fu-engine.c]: But it does not get publicly exported, I do not see any of the `fu-engine.h` methods in `Fwupd-2.0.gir`. `fu-engine.h` is also not installed to the system.

File: data/bash-completion/fwupdtool.in (hunk 3 above)
Comment (FFY00) [reply data/bash-completion/fwupdtool.in]: Riht, sorry.

File: src/fu-engine.c (hunk 1 above)
Comment (superm1) [reply src/fu-engine.c]: At least elsewhere in fu-engine.h we've been adding gtk-doc in case that stuff exports later IIRC? @hughsie ?

Comment (hughsie) [reply src/fu-engine.c]: Nth, no hard requirement. I'm not sure if bother here.

**Pull request merged**